### ISSUE TITLE
documentation error for lighttpd proxy pass, no rewrite rule

### ISSUE THREAD

**Issue by mckaygerhard** [gogits/gogs#4741]
- Gogs version (or commit ref): 0.11.29
- Git version: 2.1.0
- Operating system: Linux
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gogs.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist (usually found in `log/gogs.log`):

## Description
Use following configuration template for gogs official documentation are incorrect,due only applies to **the 1.5.X series of lighty that are not distributed** in mayor linux distributions

The portion for lighty 1.4.X seris of proxy redirection are:
```
proxy.server += ("/gogs" => ((
    "host" => "0.0.0.0",
    "port" => 3000,
)))
```
but then we need a rewrite rules equivalent to that part:
```
    proxy-core.rewrite-request = (
      "_uri" => ( "^/gogs/?(.*)" => "/$1" ),
      "Host" => ( ".*" => "localhost:3000" ),
    )
```
for the 1.4.X case of lighttpd.-.

the code documented in FAG of gogs documentation are originally:
```
server.modules  += ( "mod_proxy_backend_http" )
$HTTP["url"] =~ "^/gogs" {
    proxy-core.protocol = "http"
    proxy-core.backends = ( "localhost:3000" )
    proxy-core.rewrite-request = (
      "_uri" => ( "^/gogs/?(.*)" => "/$1" ),
      "Host" => ( ".*" => "localhost:3000" ),
    )
}
```...

**Comment by Bridouz** [comment]
Hi,

Any news on this ? I can't find any solutions for 1.4.x proxy syntax

**Comment by gstrauss** [comment]
lighttpd 1.4.46 was released today and https://redmine.lighttpd.net/projects/lighttpd/wiki/Docs_ModProxy has documentation for a new feature `proxy.header`.  The new feature does not do regex rewrites, but please take a look to see if it can be used with gogs.

**Comment by mckaygerhard** [comment]
@gstrauss that feature does not make sense.. and the same problem are happened in gitea.. seems the developers only used unoptimized and poor webservers like apache or nginx.. 

the solution its a proxy/prot redirection, see the lighty docs and forums how to redirect a directory path to a port using proxy redirection..

the other solution its to use the venenux massenkoh iso, that comes with that preconfigured.. of course binary only solution..

**Comment by gstrauss** [comment]
@mckaygerhard I don't think you tried hard enough to understand the new lighttpd feature I pointed you at.  I am one of the lighttpd developers and the new feature is cleaner and more efficient than the "proxy redirection to self" kludge.

**Comment by mckaygerhard** [comment]
@gstrauss do you see that the rewrite rules DON'T WORK IN THE LIGHTY 1.4.X SERIES that most implementations have ?

**Comment by mckaygerhard** [comment]
.. and also @gstrauss please cant then so iluminate us with right rules?

**Comment by gstrauss** [comment]
@mckaygerhard, I am having a lot of trouble understanding your broken english and your yelling.  It looks like you want the lighttpd 1.5.x (obsolete and never officially released) syntax to work with lighttpd 1.4.x.  lighttpd 1.4.x does not support that syntax and I never said that it did support that syntax.  Instead, I posted here asking that you try out lighttpd 1.4.46, and explicitly said that the new feature does not do regex rewrites.  From what you posted, lighttpd 1.4.46 did not work for you.  That's too bad.  Maybe it will help someone else.  As I said above, I don't think you tried hard enough to understand the new lighttpd feature I pointed you at.  Therefore, you may continue to yell and scream here, but I probably won't answer unless it is from someone willing to work with me, and that specifically excludes people who scream "do it for me".  Hint: if someone posts "I tried this config ..." then that demonstrates some effort.

**Comment by mckaygerhard** [comment]
do you read the gogs documentation, i try to explain without english errors, gogs have a "lighttpd" rewrite rule to make work in localhost, but that rules its not working in the all linux installations.. due the version in all linux distro are 1.4, and the gogs documentation cited lighttpd 1.5

so i open this issue, documented my experience to others and request that (with my poor contribution) the documentation rules in gogs must be changed.. or provide me any other better that can work in my lighttpd 1.4.X

also apart of all of this, if you have more experience, please tell us how we can install the gogs in localhost without call the port part, just only using http : // localhost/gogs , thanks in advance

**Comment by gstrauss** [comment]
According to what I have read in https://gogs.io/docs/intro/faqs, it should be trivial to use lighttpd 1.4.46 or later with `proxy.header` for mapping sub-paths, and if not mapping sub-paths, it is also trivial to configure earlier versions of lighttpd as a reverse-proxy. See https://redmine.lighttpd.net/projects/lighttpd/wiki/Docs_ModProxy

I tried multiple times to encourage @mckaygerhard to make any sort of effort, but @mckaygerhard  continues to feign helplessness.  If I was unclear before, allow me to repeat a key phrase:

> I probably won't answer unless it is from someone willing to work with me, and that specifically excludes people who scream "do it for me". 

@mckaygerhard: "please tell us how" is equivalent to "do it for me".  You asked me to read the gogs documentation, and I did so before my first post suggesting that lighttpd 1.4.46 had a new feature that could be used to configure gogs.  Now, I tell you the same: **RTFM**.

**Comment by mckaygerhard** [comment]
@gstrauss i make it work yet!, i open that issue to ask if are good made by me, and to please update the docs due the given info are unusefully, as you can made a search in google or like @Bridouz said...

**Comment by gstrauss** [comment]
@mckaygerhard: while lighttpd is simple enough even for most simpletons, it does not appear simple enough for you.  I ask that you not use lighttpd.

**Comment by mckaygerhard** [comment]
lighttpd its more used in large sites that need performs better rather thant very-complicated nginx or very-heaviweithg apache, like google clusters or wikipedia, hu?

when the docs changes are merged i'll close this issue

**Comment by mckaygerhard** [comment]
thanks @Unknwon,  but very long time, i just post that i'm not complety happy with that pull due the documentation must server ll lighty versions,and currently only server 1.4.46+, also documentation are inconsistent, some places have the 127.0.0.1 address and in some places have "localhost"

**Comment by Unknwon** [comment]
@mckaygerhard hi, for the part I wrote I use `localhost` (I believe :sweat_smile:). For the parts contributes by others, I'm not sure what will happen if I simply change it because I don't use those tools so I don't have expertise/knowledge. If you would like, you can send a pull request for fix everywhere, but may not actually worth doing it.

**Comment by mckaygerhard** [comment]
ok just for sure i'll send today a pull resques to use "127.0.0.1" and in that added a note that since 1.4.46+ can be use hostnames also..

**Comment by gstrauss** [comment]
Why?  My pull request includes the comment:
"# requires lighttpd 1.4.46 or later"
for the section which requires lighttpd 1.4.46 or later.

**Comment by mckaygerhard** [comment]
"requirest 1.4.46 or later" seems like dont run in lighty << 1.4.4.6, that's not true!

@gstrauss due for novice users that dont have the "lasted" you dont mention that are possible.. many users dont have lasted, just a dics with a verions older of Debian, RedHat etc, and no internet connection and just a paper and the README's

i one of those people.. i sufer too many due that bad "ambiguos" documentation... that confuses

**Comment by gstrauss** [comment]
The documentation I provided is accurate.  The section for "How do I use lighttpd with Reverse Proxy?" should work with all versions of lighttpd 1.4.x with mod_proxy.  The section "How do I set up a sub-URL with lighttpd?" is documented as "# requires lighttpd 1.4.46 or later" because the proxy.header directive did not exist in versions of lighttpd before lighttpd 1.4.46.  Since lighttpd 1.4.46 is required for proxy.header, and lighttpd 1.4.46 also supports DNS resolution at startup, 'localhost' is fine in that example.

If you are so "clever" that you "confuse" yourself by mixing-and-matching from two different sections of 4-5 line code snippets, then you probably should read directions more carefully.
